app/controllers/users_controller.rb

```diff
@@ -0,0 +1,57 @@
+class UsersController < ApplicationController
+require 'pry'
+
+  get'/users/:slug' do
+    @user = User.find_by_slug(params[:slug])
+    erb :'users/show'
+  end
+
+  get '/signup' do
+    if !logged_in?
+      erb :'users/create_user'
+    else
+      redirect to '/tweets'
+    end
+  end
+
+  post '/signup' do
+    if params[:username] == "" || params[:email] == "" || params[:password] == ""
+      redirect to '/signup'
+    else
+      @user = User.new(:username => params[:username], :email => params[:email], :password => params[:password])
```

Comment:
we could just user `User.create` here and initialize and add to the database in one method.